server/src/main/java/org/elasticsearch/script/ScoreScript.java

```diff
@@ -189,14 +192,38 @@ public void _setIndexName(String indexName) {
         this.indexName = indexName;
     }
 
+    /**
+     * Starting a name with underscore, so that the user cannot access this function directly through a script.
+     */
+    public void _setTermStats(TermStatsReader termStatsReader) {
+        this.termStatsReader = termStatsReader;
+    }
+
+    /**
+     * Accessed as _termStatistics in the painless script.
+     */
+    public TermStatsReader get_termStatistics() {
+        if (termStatsReader != null) {
+            return termStatsReader;
+        }
+
+        throw new IllegalArgumentException("_termStatistics is not available");
+    }
+
     /** A factory to construct {@link ScoreScript} instances. */
     public interface LeafFactory {
-
         /**
          * Return {@code true} if the script needs {@code _score} calculated, or {@code false} otherwise.
          */
         boolean needs_score();
 
+        /**
+         * Return {@code true} if the script needs {@code _termStatistics} calculated, or {@code false} otherwise.
+         */
+        default boolean needs_termStatistics() {
```

Comment:
ℹ️ This methods is automatically implemented by the painless script factory and return `true` if the `_termStatistics` variable is used in the script.

Reply:
I wanted to avoid reimplementing every time we have a anonymous class inheriting from ScriptScore (there is a lot of them in the tests). But I changed my mind since it is probably weird to have a default implementation for it.